engine/src/main/java/org/terasology/engine/core/modes/StateMainMenu.java

```diff
@@ -69,33 +44,15 @@ public StateMainMenu(String showMessageOnLoad) {
     @Override
     public void init(GameEngine gameEngine) {
         context = gameEngine.createChildContext();
-        CoreRegistry.setContext(context);
+        initEntityAndComponentManagers();
 
-        //let's get the entity event system running
-        EntitySystemSetupUtil.addEntityManagementRelatedClasses(context);
-        entityManager = context.get(EngineEntityManager.class);
+        createLocalPlayer(context);
 
-        eventSystem = context.get(EventSystem.class);
-        console = new ConsoleImpl(context);
-        context.put(Console.class, console);
-
-        nuiManager = new NUIManagerInternal((TerasologyCanvasRenderer) context.get(CanvasRenderer.class), context);
-        context.put(NUIManager.class, nuiManager);
+        // TODO: REMOVE this and handle refreshing of core game state at the engine level - see Issue #1127
+        new RegisterInputSystem(context).step();
 
+        nuiManager = context.get(NUIManager.class);
         eventSystem.registerEventHandler(nuiManager);
-
-        componentSystemManager = new ComponentSystemManager(context);
-        context.put(ComponentSystemManager.class, componentSystemManager);
-
-        // TODO: Reduce coupling between Input system and CameraTargetSystem,
-        // TODO: potentially eliminating the following lines. See Issue #1126
-        CameraTargetSystem cameraTargetSystem = new CameraTargetSystem();
-        context.put(CameraTargetSystem.class, cameraTargetSystem);
```

Comment:
It looks like I've removed this use of CameraTargetSystem in the main menu, but that's not really the case. It still happens within RegisterInputSystem called in this method.